### PR TITLE
Add option for JWT bearer token authentication

### PR DESCRIPTION
**Pull request by woodlee**
@civitaslearning/core-services @dkolbly Here's the mod to the common library I was mentioning in Slack earlier. Most of this is gofmt stuff; they key change is the new `UseBearerJWTAuth` field. Tyk presently uses a different bool field for each possible auth type and just evaluates them in a first-truthy-wins cascade.

### CODE REVIEW

**Comment by dkolbly** [comment]
This is basically adding a pass-thru, so that we can talk about in the meta-API (and store persistently) the `UseBearerJWTAuth` flag for a configured API?  It'll be up to `tyk` itself to make use of that configuration, right?

**Comment by derrley** [comment]
I am scared by your characterization of this as a pass thru! Now I need to
take a look.

**Comment by woodlee** [comment]
@dkolbly Correct. @derrley This is a seeming shortcoming of Tyk. Right now, the meta-API configuration is not at all dynamic. Different fields for every setting!

**Comment by kristenhunter** [comment]
For new users of github: this is a great candidate for adding a ?w=1 at the end of the url!  https://github.com/civitaslearning/tykcommon/pull/1/files?w=1 I

File: api_defintitions.go [hunk 1]
```diff
@@ -111,24 +112,24 @@ type APIDefinition struct {
...
+	DontSetQuotasOnCreate     bool                   `mapstructure:"dont_set_quota_on_create" bson:"dont_set_quota_on_create" json:"dont_set_quota_on_create"`
+	ExpireAnalyticsAfter      int64                  `mapstructure:"expire_analytics_after" bson:"expire_analytics_after" json:"expire_analytics_after"` // must have an expireAt TTL index set (http://docs.mongodb.org/manual/tutorial/expire-data/)
+	RawData                   map[string]interface{} `bson:"raw_data,omitempty" json:"raw_data,omitempty"`                                               // Not used in actual configuration, loaded by config for plugable arc
 }
 
 // Clean will URL encode map[string]struct variables for saving
 func (a *APIDefinition) EncodeForDB() {
 	new_version := make(map[string]VersionInfo)
-	for k, v := range(a.VersionData.Versions) {
+	for k, v := range a.VersionData.Versions {
```
Comment (kristenhunter) [inline api_defintitions.go]: why this change?

Comment (woodlee) [inline api_defintitions.go]: Eh, Go is weird(/awesome) this way in that it has a formatter (gofmt) which, if you use it, will bring everything into canonical Go style. This was simply a change gofmt made, and though I can't point to the style guidelines that say "don't use parentheses here", I tend to just trust it.

Comment (kristenhunter) [inline api_defintitions.go]: Works for me!

**Comment by woodlee** [comment]
Anyone ready to :shipit: me on this one? @derrley are you still wanting to look/think more on this one?

**Comment by dkolbly** [comment]
Looks good to me :ship:
